pkg/router/f5/f5.go

```diff
@@ -459,6 +459,35 @@ func (f5 *f5LTM) delete(url string, result interface{}) error {
 }
 
 //
+// iControl REST resource helper methods.
+//
+
+// iControlResourceUriPath returns a fully qualified `escaped` resource URI path component that can be used in the iControl REST calls.
+func (f5 *f5LTM) iControlResourceUriPath(resourceName string) string {
+	resourcePath := path.Join(f5.partitionPath, resourceName)
+	return escapeiControlUriPath(resourcePath)
+}
+
+// iControlVserverUriPath returns a fully qualified `escaped` Virtual Server path that can be used in the iControl REST calls.
+func (f5 *f5LTM) iControlVserverUriPath(vserverName string) string {
+	// Build the fully qualified Virtual Server name.
+	// Note: Although the configuration is stored inside a partition path, the F5 virtual server
+	// belongs to the partition's top-level folder (directory).
+	// Example for a vserver call foo-server, this would be
+	//    /Common/foo-server for the default /Common path
+	// and
+	//    /ose/foo-server for a custom /ose/v3/config path.
+	partitionDir := path.Join("/", F5DefaultPartitionPath)
+
+	pathComponents := strings.Split(f5.partitionPath, "/")
+	if len(pathComponents) > 1 && len(pathComponents[1]) > 0 {
+		partitionDir = path.Join("/", pathComponents[1])
+	}
```

Comment:
Is it possible (or should it be possible) for `f5.partitionPath` to be "/"? If not, I don't see the point of lines 480 and 483—why not simply take `f5.partitionPath` and extract the first path component?

I think we need to revisit, possibly revise, and document the permissible values for `partitionPath` under `f5LTMCfg`:

* State clearly whether or not `partitionPath` can be set to "/" and, if so, what the value "/" means.  Does it just mean "use the default", i.e., the value of `F5DefaultPartitionPath`, i.e., "/Common"?
* Point out that `f5.partitionPath` can have more than 1 component—for example, the 3-component "/ose/v3/config" in your above comment.
* Perhaps list which resources go under partitions (is it all of them?) and note exceptions (vservers and any others that we use).  Maybe it's redundant, but I am surprised to learn that vservers can only exist under top-level partitions, and it may be worth documenting that fact more prominently.  I'm also curious whether partitions apply to data-groups, certificates and keys, or SSL profiles (I noticed you didn't change the code to specify the partition for any of these resources), and such a comment would clear that up.